src/Nri/Ui/SegmentedControl/V3.elm

```diff
@@ -0,0 +1,124 @@
+module Nri.Ui.SegmentedControl.V3 exposing (Config, Icon, Option, styles, view)
+
+{-|
+
+@docs Config, Icon, Option, styles, view
+
+-}
+
+import Accessibility exposing (..)
+import Accessibility.Role as Role
+import Css exposing (..)
+import Css.Foreign exposing (Snippet, adjacentSiblings, children, class, descendants, each, everything, media, selector, withClass)
+import Html
+import Html.Attributes
+import Html.Events
+import Nri.Ui.Colors.Extra exposing (withAlpha)
+import Nri.Ui.Colors.V1 as Colors
+import Nri.Ui.CssFlexBoxWithVendorPrefix as FlexBox
+import Nri.Ui.Fonts.V1 as Fonts
+import Nri.Ui.Icon.V1 as Icon
+import Nri.Ui.Styles.V1
+import View.Extra
+
+
+{-| -}
+type alias Config a msg =
+    { onClick : a -> msg
+    , options : List (Option a)
+    , selected : a
+    }
+
+
+{-| -}
+type alias Option a =
+    { value : a
+    , icon : Maybe Icon
+    , label : String
+    , id : String
+    }
+
+
+{-| -}
+type alias Icon =
+    { alt : String
+    , icon : Icon.IconType
+    }
+
+
+{-| -}
+view : Icon.Assets r -> Config a msg -> Html.Html msg
```

Comment:
💭 I'm concerned that this function requires all the icon assets rather than just those icons used in this particular function. In CCS this would see us adding a lot of icons of which only a few would ever be used. We could 'fake' those assets by passing empty strings as paths, but then we wouldn't see a compile time error when we start using a previously unused icon.

 I think this is something we should address in a follow-up PR, so that we can merge this one.

Reply:
Yes, that does seem like a pain. Since the segmented control can use any icon, what other choice do we have?

But also, CCS will have to deal with all the assets at some point. Might as well be now 😅.

Reply:
That's fair!

Although I realized there's a bigger problem. Because `IconType` is a union type, every icon we add will require a major version bump of the package, or a new version of `Nri.Ui.Icon.Vx`. Yikes either way. I have a suggestion with an alternative approach in #23.